integrations/extensions/docs/elasticsearch-install-and-setup/text_embedding_deploy_and_use.md

```diff
@@ -0,0 +1,189 @@
+curl -X PUT "${ES_URL}/_ingest/pipeline/text-embeddings" \
+  -u "${ES_USER}:${ES_PASSWORD}" --cacert "${ES_CACERT}"\
+  -H 'Content-Type: application/json' -d '{
+  "description": "Text embedding pipeline",
+  "processors": [
+    {
+      "inference": {
+        "model_id": "intfloat__multilingual-e5-small",
+        "target_field": "text_embedding",
+        "field_map": {
+          "text": "text_field"
+        }
+      }
+    }
+  ],
+  "on_failure": [
+    {
+      "set": {
+        "description": "Index document to '\''failed-<index>'\''",
+        "field": "_index",
+        "value": "failed-{{{_index}}}"
+      }
+    },
+    {
+      "set": {
+        "description": "Set error message",
+        "field": "ingest.failure",
+        "value": "{{_ingest.on_failure_message}}"
+      }
+    }
+  ]
+}'
+```
+
+You can verify that the `text-embeddings` ingest pipeline was created by locating it in the list of your ingest pipelines on Kibana http://localhost:5601/app/management/ingest/ingest_pipelines
+
+## Create a mapping for the destination index containing the embeddings
+Then run the command below to create the mappings of the destination index called `wa-docs-with-embeddings`:
+```bash
+curl -X PUT "${ES_URL}/wa-docs-with-embeddings" \
+  -u "${ES_USER}:${ES_PASSWORD}" --cacert $ES_CACERT \
+  -H 'Content-Type: application/json' -d '{
+  "mappings": {
+    "properties": {
+      "text_embedding.predicted_value": {
+        "type": "dense_vector",
+        "dims": 384,
+        "index": true,
+        "similarity": "cosine"
+      },
+      "text": {
+        "type": "text"
+      }
+    }
+  }
+}'
+```
+
+* `text_embedding.predicted_value` is the field where the ingest processor stores the embeddings
+* `dims` is the embedding size of the deployed model which is 384 for the `intfloat/multilingual-e5-small` model we are using here
+
+## Create the text embeddings
+Run the `text-embeddings` ingest pipeline to reindex the data to the `wa-docs-with-embeddings` index
+```bash
+curl -X POST "${ES_URL}/_reindex?wait_for_completion=false" \
+  -u "${ES_USER}:${ES_PASSWORD}" --cacert $ES_CACERT \
+  -H 'Content-Type: application/json' -d '{
+  "source": {
+    "index": "wa-docs",
+    "size": 50
+  },
+  "dest": {
+    "index": "wa-docs-with-embeddings",
+    "pipeline": "text-embeddings"
+  }
+}'
+```
+
+This command will return a task id that looks like this:
+```json
+{"task":<task-id>}
+```
+
+The reindexing process can take around 10 minutes. You can use the task id that is returned above to check the status of the process.
+
+```bash
+curl -X GET "${ES_URL}/_tasks/<task-id>" \
+  -u "${ES_USER}:${ES_PASSWORD}" --cacert $ES_CACERT
+```
+
+Once the process is complete, you should see `wa-docs-with-embeddings` in the list of your indices http://localhost:5601/app/enterprise_search/content/search_indices
+
+## Run semantic search
+After the dataset has been enriched with vector embeddings, you can query the data using semantic search. 
```

Comment:
I'm getting an error running this. I'll pause my review here.

`{"error":{"root_cause":[{"type":"query_shard_exception","reason":"failed to create query: [knn] queries are only supported on [dense_vector] fields","index_uuid":"E50WgDeoS6ONscKjyHkO3Q","index":"ewayne-wa-docs-with-embeddings"}],"type":"search_phase_execution_exception","reason":"all shards failed","phase":"dfs","grouped":true,"failed_shards":[{"shard":0,"index":"ewayne-wa-docs-with-embeddings","node":"usod4PNbQU6r7LrzoShkpw","reason":{"type":"query_shard_exception","reason":"failed to create query: [knn] queries are only supported on [dense_vector] fields","index_uuid":"E50WgDeoS6ONscKjyHkO3Q","index":"ewayne-wa-docs-with-embeddings","caused_by":{"type":"illegal_argument_exception","reason":"[knn] queries are only supported on [dense_vector] fields"}}}]},"status":400}`

Reply:
This is interest! I am able to replicate the error. This happens when [wa-docs-100](http://localhost:5601/app/enterprise_search/content/search_indices/wa-docs-100/) is used as the source index. I didn't run into this because I used the original [wa-docs](http://localhost:5601/app/enterprise_search/content/search_indices/wa-docs/) index as the source. 

The only difference I see between these two indices is that in wa-docs all columns are type `text` while in wa-docs-100 the title column is automatically set to the `keyword` type and I can't seem to modify it. @sne3091 @zach-shu do you know how wa-docs-100 was created?

![image](https://github.com/watson-developer-cloud/assistant-toolkit/assets/5916032/239c0e47-3801-4d4d-9e04-46e0bfa63716)

![image](https://github.com/watson-developer-cloud/assistant-toolkit/assets/5916032/f5753901-e6dc-4d93-b4aa-0dfdc2900b2c)

Reply:
This issue is not encountered anymore so closing this.